src/write/mod.rs

```diff
@@ -56,24 +57,53 @@ macro_rules! define_section {
     };
 }
 
+macro_rules! define_id {
+    ($name:ident, $docs:expr) => {
+        #[doc=$docs]
+        #[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+        pub struct $name {
+            base_id: BaseId,
```

Comment:
Maybe add `#[cfg(debug_assertions)]`. That would save a few bytes.

Reply:
I'll try changing `BaseId`'s type to `()` in release mode.